contract/contracts/PixelAvatars.sol

```diff
@@ -30,21 +37,22 @@ contract PixelAvatars is
         return baseURI;
     }
 
-    function setBaseURI(string memory _newBaseURI) public onlyOwner {
+    function setBaseURI(string memory _newBaseURI) external onlyOwner {
         baseURI = _newBaseURI;
+        emit BaseURIUpdated(baseURI, _newBaseURI);
```

Comment:
You need to emit the event before the state mutation, otherwise what it's gonna emit is the same values and wouldn't get the old value.

Suggested change is to switch line 41 and line 42